promql/engine.go

```diff
@@ -516,7 +521,8 @@ func (ng *Engine) populateIterators(ctx context.Context, s *EvalStmt) (storage.Q
 		switch n := node.(type) {
 		case *VectorSelector:
 			n.series, err = expandSeriesSet(querier.Select(n.LabelMatchers...))
-			if err != nil {
+			// No storage backend has returned the series so can't continue the evaluation
```

Comment:
We shouldn't be special casing the number of series returned. It's completely valid for a storage to return no series for a selector.